internal/report/source.go

```diff
@@ -270,15 +280,21 @@ func newSourcePrinter(rpt *Report, obj plugin.ObjTool, sourcePath string) *sourc
 				sp.prettyNames[line.Function.SystemName] = line.Function.Name
 			}
 
-			cum[loc.Address] += value
+			addr := loc.Address
+			if addr == 0 {
+				// Some profiles are missing valid addresses.
+				addr = sp.synth.address(loc)
+			}
+
+			cum[addr] += value
```

Comment:
This is pre-existing and doesn't have to be changed (if it needs to be changed) in this CL, but: do we handle recursion correctly here? It feels that we'll double count the cumulative time if the same address is present multiple times in the stack.

Reply:
I think you are right. Let me look at that in a separate change.

Reply:
Hate to butt in here, but is there a chance that this is related to #720?

I'm not familiar with this code, but a flavor of #720 manifested for me with the following conditions:
 - using `weblist` instead of `list`
 - providing the binary for disassembly

and had the following behavior:
 - in a sample function that had 4 caller locations, the count was 5x what the expected value (e.g. that returned by `list` or `weblist` without the binary) was.
 - the printed assembly was printed 5 times
<img width="1265" alt="image" src="https://user-images.githubusercontent.com/5506073/229330897-ce46d10c-d65b-4d5f-bc0c-bfbac8619bcc.png">


As far as I can tell, that fits the bill for the code in question here: the call tree for this function is only invoked for web printing and not the CLI interactions